### ISSUE TITLE
[java] False-positive UseProperClassLoader

### ISSUE THREAD

**Issue by oliveryasuna** [pmd/pmd#4711]
**Affects PMD Version:** LATEST

**Rule:** [UseProperClassLoader](https://docs.pmd-code.org/pmd-doc-6.55.0/pmd_rules_java_errorprone.html#useproperclassloader)

**Description:**

Reports Spring's `DataSourceProperties#getClassLoader()` as a UseProperClassLoader violation.

**Code Sample demonstrating the issue:**

```java
@Bean
@Autowired
public DataSourceWithProperties dataSource(final DataSourceProperties properties, final JdbcConnectionDetails connectionDetails) {
  return DataSourceBuilder.create(properties.getClassLoader())
      .type(HikariDataSource.class)
      .driverClassName(connectionDetails.getDriverClassName())
      .url(connectionDetails.getJdbcUrl())
      .username(connectionDetails.getUsername())
      .password(connectionDetails.getPassword())
      .build();
}
```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** Gradle

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0 as long as Spring Boot is properly included in the auxclasspath.
